packages/app/src/cli/services/app-logs/utils.ts

```diff
@@ -161,6 +164,42 @@ export const handleFetchAppLogsError = async (
   return {retryIntervalMs, nextJwtToken}
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const toFormattedAppLogJson = (appLog: AppLogData, appLogPayload: AppLogPayload | any): string => {
+  const {cursor: _, ...appLogWithoutCursor} = appLog
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const toSaveData: any = camelcaseKeys(
+    {
+      ...appLogWithoutCursor,
+      payload: appLogPayload,
+    },
+    {deep: true},
+  )
+
+  if (appLogPayload instanceof FunctionRunLog) {
+    toSaveData.payload.logs = appLogPayload.logs.split('\n').filter(Boolean)
+  }
+
+  return JSON.stringify(toSaveData, null, 2)
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const parseAppLogPayload = (payload: string, logType: string): AppLogPayload | any => {
+  const parsedPayload = camelcaseKeys(JSON.parse(payload), {deep: true})
+
+  if (logType === LOG_TYPE_FUNCTION_RUN) {
+    return new FunctionRunLog(parsedPayload)
+  } else if (logType === LOG_TYPE_RESPONSE_FROM_CACHE) {
+    return new NetworkAccessResponseFromCacheLog(parsedPayload)
+  } else if (logType === LOG_TYPE_REQUEST_EXECUTION_IN_BACKGROUND) {
+    return new NetworkAccessRequestExecutionInBackgroundLog(parsedPayload)
+  } else if (logType === LOG_TYPE_REQUEST_EXECUTION) {
+    return new NetworkAccessRequestExecutedLog(parsedPayload)
+  } else {
+    return parsedPayload
+  }
```

Comment:
This looks good 👍 
We should use it in our `poll-app-logs` as well I think.
https://github.com/Shopify/cli/blob/6a6ecaec0e5ba4d411d8c4fe368b83293f64d40d/packages/app/src/cli/services/app-logs/dev/poll-app-logs.ts#L81-L85

Lets just open another issue with this comment once this has shipped to avoid delaying this PR 🚤